### DISCUSSION TITLE
Trajectory creation from Blender Export

### DISCUSSION THREAD

**Discussion by stikkx** [USC-ACTLab/crazyswarm#312] [Q&A]
Hello, 

i have a trajectory of 12 CFs build in Blender. I have a export written in python to CSV. But with format is the best ? I think waypoints are to much to transmit while the CFs are flying. And to use the uav_trajectory script is in my opinion unnecessary since I already have the exact trajectory. Can't I create atrajectory directly from the cooardinates with time information (up to 24/second) and load it on the CFs ? Which format should it have ?

Thanks a lot!

**Comment by jpreiss** [comment]
Please explain more precisely what you mean by "a trajectory directly from the coordinates with time information". What format are you getting out of Blender?

**Comment by stikkx** [comment]
I get the time in seconds or milliseconds and the position x,y,z,yaw in meters per frame from the animation in Blender.

**Comment by whoenig** [comment]
I think the best would be to export this data to a csv-file and then convert to trajectories by following https://github.com/whoenig/uav_trajectories#generate-trajectory-given-time-position-pairs.

**Comment by stikkx** [comment]
I tried that, but the result is a bit laggy with --sim parameter and without the CF is flying very instabel.

Which format is the trajectory the script puts out?

Thank you!

**Comment by jpreiss** [comment]
Our trajectory system uses piecewise polynomials. The .csv file contains the polynomial coefficients. I don't think the format is documented, but you can inspect [the parsing code](https://github.com/USC-ACTLab/crazyswarm/blob/dfd69b44fcb8fe78f11fa3d76e1a35e4cea715c8/ros_ws/src/crazyswarm/scripts/uav_trajectory.py#L96-L99) to understand it.

If you are designing smooth trajectories with Blender, it is likely that Blender uses some spline format internally to represent the trajectory. It should be possible to do a "lossless" conversion to our polynomial format. That would be the ideal workflow, if you can figure it out.

**Comment by jpreiss** [comment]
@stikkx did you make any progress here?

**Comment by stikkx** [comment]
hi jpreiss, i hope i could make some progress in the next week. But atm i had no time to do..
